ntoskrnl/kd/kdio.c

```diff
@@ -184,15 +184,15 @@ KdpLoggerThread(PVOID Context)
 
         if (end > beg)
         {
-            NtWriteFile(KdpLogFileHandle, NULL, NULL, NULL, &Iosb,
+            ZwWriteFile(KdpLogFileHandle, NULL, NULL, NULL, &Iosb,
```

Comment:
I'm not strongly against this but also not super in favor of it.
An alternative that doesn't kill the optimization would be to `ASSERT(ExGetPreviousMode() == KernelMode)` to show that using Nt* is safe in these functions.

Reply:
The logger thread is a system thread so it should normally run in kernel mode context only. (So the Nt* functions are safe to call.)

Reply:
> I'm not strongly against this but also not super in favor of it.

Yeah, I wondered too.

> `ASSERT(ExGetPreviousMode() == KernelMode)`

Replaced :-)

Reply:
> The logger thread is a system thread

Yes, this commit is only about being explicit/extra-safe.

Reply:
Well then you don't need that. It's not an "extra-safe" feature, but it's an under-optimization (now going through the system-call table etc.)

Reply:
> Well then you don't need that. It's not an "extra-safe" feature, but it's an under-optimization (now going through the system-call table etc.)

I am not sure what you want to debate further: I think my answer to Thomas and my commit update are enough here.